frontend/src/app/homepage-offer/offers.service.ts

```diff
@@ -1,38 +1,51 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-
 import 'rxjs/add/operator/map';
 import { Observable } from 'rxjs/Observable';
 
 import { environment } from '../../environments/environment';
-import { Offer } from './offers.model';
+import { ApiOffer, AppOffer } from './offers.model';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
 
 
 @Injectable()
 export class OffersService {
   private url = `${environment.apiRoot}/offers/`;
+  private _offer$ = new BehaviorSubject<ApiOffer>(null);
```

Comment:
I dug deep into your code and I kind of understand what you were aiming to do here. However, it is not a solution that works and although it works in organization service, for now offers service works a little bit differently. I will give you the solution to the problem you had in offer-details.component.